### ISSUE TITLE
langchain models throw `LLMNotFoundError` as a consequence of pydantic's validator

### ISSUE THREAD

**Issue by mspronesti** [gventuri/pandas-ai#509]
### 🐛 Describe the bug

This issue is a generalization of #496, to better discuss how to tackle it.

In few: the validation performed at [this line](https://github.com/gventuri/pandas-ai/blob/main/pandasai/helpers/df_config.py#L30) will always fail for langchain's LLMs because [`load_llm`](https://github.com/gventuri/pandas-ai/blob/main/pandasai/smart_datalake/__init__.py#L150-L169) - where the conversion is performed - is always called after creating the `Config` object [here](https://github.com/gventuri/pandas-ai/blob/main/pandasai/smart_datalake/__init__.py#L75). 

Possible solutions:
- we perform the conversion at the level of the `Config` (simplest thing to do, in my opinion)
- we somehow load the llm before creating a new `Config` object. This only makes sense when `config` is a raw dictionary, and still forces the user to convert the llm on their own if they prefer to create a `Config` object to pass to `SmartDataFrame`'s constructor (which in fact expects to receive an object of type `pandasai.Config`)

**Comment by gventuri** [comment]
@mspronesti thanks a lot for opening a dedicate issue.

So if I understand correctly, you mean something like this:

```python@validator("llm")
    def validate_llm(cls, llm):
        if llm is None:
            raise LLMNotFoundError("LLM is required")
        elif isinstance(llm, str):
            llm = LangchainLLM.load_llm(llm)
        elif not isinstance(llm, (LLM, LangchainLLM)):
            raise ValueError("Invalid LLM type")
        return llm

**Comment by gventuri** [comment]
@mspronesti what about this: https://github.com/gventuri/pandas-ai/pull/512?

**Comment by gventuri** [comment]
Closing, as the issue has been addressed with #512
